pkg/ssh/ssh.go

```diff
@@ -2,46 +2,124 @@ package ssh
 
 import (
 	"context"
-	"log"
+	"flag"
+	"fmt"
+	"net/url"
 	"os"
 	"os/exec"
+	"path"
+	"strings"
 	"time"
 
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
 	"github.com/grafana/dskit/services"
+	"github.com/grafana/pdc-agent/pkg/pdc"
 )
 
+// Config represents all configurable properties of the ssh package.
 type Config struct {
-	Args []string
+	Args []string // deprecated
+
+	KeyFile               string
+	SSHFlags              []string // Additional flags to be passed to ssh(1). e.g. --ssh-flag="-vvv" --ssh-flag="-L 80:localhost:80"
+	ForceKeyFileOverwrite bool
+	Port                  int
+	PDC                   pdc.Config
+	LegacyMode            bool
+	URL                   *url.URL
+}
+
+const forceKeyFileOverwriteUsage = `If enabled, the pdc-agent will regenerate an SSH key pair and request a new
+certificate to use whem establishing an SSH tunnel.
+
+If disabled, pdc-agent will use existing SSH keys and only request a new SSH
+certificate when the existing one is expired. If no SHH keys exist, it will
+generate a pair and request a certificate.`
+
+// DefaultConfig returns a Config with some sensible defaults set
+func DefaultConfig() *Config {
+
+	return &Config{
+		Port:    22,
+		PDC:     pdc.Config{},
+		KeyFile: "~/.ssh/gcloud_pdc",
+	}
+}
```

Comment:
I think the tilde isn't recognised in go.

```suggestion


func DefaultConfig() *Config {
        root, err := os.UserHomeDir()
	if err != nil {
		// ??? Maybe lets use the current dir? 
		root = ""
	}
	return &Config{
		Port:    22,
		PDC:     pdc.Config{},
		KeyFile: filepath.Join(root, ".ssh/gcloud_pdc"),
	}
}
```

Reply:
thanks I missed this. I think using the current dir is a good idea

Reply:
I think we should still use the user Home directory (~/.ssh/gcloud_pdc ). I can think of a few reasons:
- The current user generally has write access to the Home directory. 
- Certain users will want to lock this directory down. Making the path relative makes this less deterministic.
-